File: modules/agent/src/serializers/note.py
"""Serializers for note-related models."""


def note_to_response(note) -> dict:
    """Convert Note model to dictionary."""
    creator = note.creator
    created_by_name = None
    created_by_email = None
    individual_id = None
    if creator:
        first = creator.first_name or ""
        last = creator.last_name or ""
        created_by_name = f"{first} {last}".strip() or None
        created_by_email = creator.email
        individual_id = creator.individual_id

    return {
        "id": note.id,
        "tenant_id": note.tenant_id,
        "entity_type": note.entity_type,
        "entity_id": note.entity_id,
        "content": note.content,
        "created_by": note.created_by,
        "created_by_name": created_by_name,
        "created_by_email": created_by_email,
        "individual_id": individual_id,
        "created_at": note.created_at.isoformat() if note.created_at else None,
        "updated_at": note.updated_at.isoformat() if note.updated_at else None,
    }

File: modules/agent/internal/serializers/note.go
package serializers

import (
	"time"

	"agent/internal/models"
)

// NoteResponse represents a note
type NoteResponse struct {
	ID         int64     `json:"id"`
	TenantID   int64     `json:"tenant_id"`
	EntityType string    `json:"entity_type"`
	EntityID   int64     `json:"entity_id"`
	Content    string    `json:"content"`
	CreatedBy  int64     `json:"created_by"`
	UpdatedBy  int64     `json:"updated_by"`
	CreatedAt  time.Time `json:"created_at"`
	UpdatedAt  time.Time `json:"updated_at"`
}

// NoteToResponse converts a model to response
func NoteToResponse(m *models.Note) NoteResponse {
	return NoteResponse{
		ID:         m.ID,
		TenantID:   m.TenantID,
		EntityType: m.EntityType,
		EntityID:   m.EntityID,
		Content:    m.Content,
		CreatedBy:  m.CreatedBy,
		UpdatedBy:  m.UpdatedBy,
		CreatedAt:  m.CreatedAt,
		UpdatedAt:  m.UpdatedAt,
	}
}

// NotesToResponse converts a slice of models to responses
func NotesToResponse(models []models.Note) []NoteResponse {
	resp := make([]NoteResponse, len(models))
	for i := range models {
		resp[i] = NoteToResponse(&models[i])
	}
	return resp
}
